tools/internal/shell/interface.go

```diff
@@ -0,0 +1,62 @@
+package shell
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+//
+// Runner is an interface for running shell commands. It exists to
+// support mocking shell commands in unit tests.
+//
+// https://joshrendek.com/2014/06/go-lang-mocking-exec-dot-command-using-interfaces/
+//
+type Runner interface {
+	Run(cmd Command) error
+}
+
+// Error represents an error encountered running a shell command
+type Error struct {
+	Command Command
+	Err     error
+}
+
+// Command encapsulates a shell command
+type Command struct {
+	Prog        string   // Prog Main CLI program to execute
+	Args        []string // Args Arguments to pass to program
+	Dir         string   // Dir Directory where command should be run
+	Env         []string // Env List of environment variables, eg []string{ "FOO=BAR", "BAZ=QUUX" }, to set when executing
+	PristineEnv bool     // PristineEnv When true, set only supplied Env vars without inheriting current process's env vars
+}
+
+// Error generates a user-friendly error message for failed shell commands
+func (e *Error) Error() string {
```

Comment:
are we using `err` for the standard package `error` and `e` for a custom `Error` struct?

Reply:
Eh. I guess I typically use short variable names for receivers (which is why I used a short one here) and `err` during regular error-handling conditions. Happy to change the use here, though.

Reply:
Done!